packages/jest-resolve/src/index.ts

```diff
@@ -377,28 +377,42 @@ class Resolver {
           // Note: once a moduleNameMapper matches the name, it must result
           // in a module, or else an error is thrown.
           const matches = moduleName.match(regex);
-          const updatedName = matches
-            ? mappedModuleName.replace(
-                /\$([0-9]+)/g,
-                (_, index) => matches[parseInt(index, 10)],
-              )
-            : mappedModuleName;
-
-          const module =
-            this.getModule(updatedName) ||
-            Resolver.findNodeModule(updatedName, {
-              basedir: dirname,
-              browser: this._options.browser,
-              extensions,
-              moduleDirectory,
-              paths,
-              resolver,
-              rootDir: this._options.rootDir,
-            });
+          const mapModuleName = matches
+            ? (moduleName: string) =>
+                moduleName.replace(
+                  /\$([0-9]+)/g,
+                  (_, index) => matches[parseInt(index, 10)],
+                )
+            : (moduleName: string) => moduleName;
+
+          const possibleModuleNames = Array.isArray(mappedModuleName)
```

Comment:
Can we make it into an array in `normalize`?

Thinking about it, that might be a breaking change... Hmm

Reply:
When an error happens we need to properly format error message based on whether it was set as an array or string in configuration. If we normalize it, we will probably lose that information and not be able to properly format error message

Reply:
Wouldn't be too bad to show `[path]` even though the user configured `path` I guess, but probably not worth it also given the breaking change risk. I'd be happy as is - however one thing I do want to verify when I have some time is whether the extra work of `Array.isArray`, wrapping `[mappedModuleName]` etc. has a perf impact. I doubt it's significant, but `jest-resolve` is quite a hot path so better safe than sorry

Reply:
could normalize to an array in the constructor

Reply:
Not sure I understood that.  Is there anything for me to do apart from updating docs?

Reply:
@jeysal Aren't resolved module paths cached?

Reply:
```ts
this._possibleModuleNames = Array.isArray(mappedModuleName) ? mappedModuleName : [mappedModuleName];
```
in the constructor of this class

Reply:
@SimenB The problem is we need both the original config entry for the error message and the array one to iterate over here. As long as we don't want to compromise on the correctness of the error message I think this is as good as it gets 😅

Reply:
> @jeysal Aren't resolved module paths cached?

Even with caching often running a single test will need to resolve a large dependency tree, just want to be on the safe side as far as perf goes 😄

Reply:
the error message can be passed the original config, but in the happy case it won't have to check if array or not